### PR TITLE
executor: migrate test-infra to testify for sample_test.go

### LINKED ISSUE #28606: migrate test-infra to testify for `sample_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28606

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/sample_test.go [hunk 1]
```diff
@@ -18,67 +18,62 @@ import (
...
+		err := store.Close()
+		require.NoError(t, err)
+	}
+	return
 }
 
-func (s *testTableSampleSuite) TestTableSampleBasic(c *C) {
-	tk := s.initSampleTest(c)
+func TestTableSampleBasic(t *testing.T) {
+	_, tk, clean := createSampleStore(t)
```
Comment (hawkingrei) [inline executor/sample_test.go:73]: should use 

```
store, clean := testkit.CreateMockStore(t)
defer clean()
tk := testkit.NewTestKit(t, store)
```

File: executor/sample_test.go [hunk 2]
```diff
@@ -105,14 +100,15 @@ func (s *testTableSampleSuite) TestTableSampleBasic(c *C) {
...
+func TestTableSampleMultiRegions(t *testing.T) {
+	_, tk, clean := createSampleStore(t)
+	defer clean()
 	tk.MustExec("create table t (a int) shard_row_id_bits = 2 pre_split_regions = 2;")
 	for i := 0; i < 100; i++ {
 		tk.MustExec("insert into t values (?);", i)
 	}
 	rows := tk.MustQuery("select * from t tablesample regions();").Rows()
-	c.Assert(len(rows), Equals, 4)
+	require.Equal(t, 4, len(rows))
```
Comment (hawkingrei) [inline executor/sample_test.go:111]: use ```require.Len```.

ditto other.

File: executor/sample_test.go (hunk 1 above)
Comment (feitian124) [reply executor/sample_test.go:73]: thank your for review.
i see the original init logic in `SetUpSuite` almost the same as `testkit.CreateMockStore(t)` except the latter one missing flag `mockTikv`, which means after migrate the flag is not used and alwasy use mockStore. 
Is this acceptable? I am not sure this part.

File: executor/sample_test.go (hunk 2 above)
Comment (feitian124) [reply executor/sample_test.go]: fixed

File: executor/sample_test.go (hunk 1 above)
Comment (hawkingrei) [reply executor/sample_test.go]: Yes, CreateMockStore can support the ```mockTikv```.  You can replace it.

Comment (feitian124) [reply executor/sample_test.go]: fxied.
use `testkit.CreateMockStore` but keep `createSampleStore` to remove duplicated code like create and use database.

**Review by hawkingrei — APPROVED** [review]

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**